mne/simulation/raw.py

```diff
@@ -239,11 +258,8 @@ def simulate_raw(raw, stc, trans, src, bem, cov='simple',
     src = _ensure_src(src, verbose=False)
     if isinstance(bem, string_types):
         bem = read_bem_solution(bem, verbose=False)
-    if isinstance(cov, string_types):
-        if cov == 'simple':
-            cov = make_ad_hoc_cov(info, verbose=False)
-        else:
-            cov = read_cov(cov, verbose=False)
+    if cov is not None:
+        cov = _check_cov(info, cov)
```

Comment:
don't nest this, instead handle the `if cov is None` case in `_check_cov` (keeps all checking logic in one place)